commit 7e85c72969b17eed698c3e23674f424bf345aceb
Author: Jason Cooper <jason@lakedaemon.net>
Date:   Tue Sep 14 09:45:38 2010 -0400

    staging: brcm80211: fix "ERROR: do not initialise statics to 0 or NULL"
    
    Signed-off-by: Jason Cooper <jason@lakedaemon.net>
    Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

--- a/drivers/staging/brcm80211/sys/wl_mac80211.c
+++ b/drivers/staging/brcm80211/sys/wl_mac80211.c
@@ -128,7 +128,7 @@ static int wl_linux_watchdog(void *ctx);
 	FIF_OTHER_BSS | \
 	FIF_BCN_PRBRESP_PROMISC)
 
-static int wl_found = 0;
+static int wl_found;
 
 struct ieee80211_tkip_data {
 #define TKIP_KEY_LEN 32
@@ -203,17 +203,17 @@ module_param(phymsglevel, int, 0);
 #endif				/* WLC_HIGH_ONLY */
 #endif				/* BCMDBG */
 
-static int oneonly = 0;
+static int oneonly;
 module_param(oneonly, int, 0);
 
-static int piomode = 0;
+static int piomode;
 module_param(piomode, int, 0);
 
-static int instance_base = 0;	/* Starting instance number */
+static int instance_base;	/* Starting instance number */
 module_param(instance_base, int, 0);
 
 #if defined(BCMDBG)
-static char *macaddr = NULL;
+static char *macaddr;
 module_param(macaddr, charp, S_IRUGO);
 #endif
 
--- a/drivers/staging/brcm80211/util/bcmutils.c
+++ b/drivers/staging/brcm80211/util/bcmutils.c
@@ -29,7 +29,7 @@
 
 #ifdef WLC_LOW
 /* nvram vars cache */
-static char *nvram_vars = NULL;
+static char *nvram_vars;
 static int vars_len = -1;
 #endif				/* WLC_LOW */
 
--- a/drivers/staging/brcm80211/util/hnddma.c
+++ b/drivers/staging/brcm80211/util/hnddma.c
@@ -48,7 +48,7 @@
 #define rxd64		dregs.d64_u.rxd_64
 
 /* default dma message level (if input msg_level pointer is null in dma_attach()) */
-static uint dma_msg_level = 0;
+static uint dma_msg_level;
 
 #define	MAXNAMEL	8	/* 8 char names */
 
--- a/drivers/staging/brcm80211/util/hndpmu.c
+++ b/drivers/staging/brcm80211/util/hndpmu.c
@@ -1840,7 +1840,7 @@ uint32 BCMINITFN(si_pmu_mem_clock) (si_t
 /* Measure ILP clock frequency */
 #define ILP_CALC_DUR	10	/* ms, make sure 1000 can be divided by it. */
 
-static uint32 ilpcycles_per_sec = 0;
+static uint32 ilpcycles_per_sec;
 
 uint32 BCMINITFN(si_pmu_ilp_clock) (si_t *sih, osl_t *osh)
 {
--- a/drivers/staging/brcm80211/util/nvram/nvram_ro.c
+++ b/drivers/staging/brcm80211/util/nvram/nvram_ro.c
@@ -38,7 +38,7 @@ typedef struct _vars {
 
 #define	VARS_T_OH	sizeof(vars_t)
 
-static vars_t *vars = NULL;
+static vars_t *vars;
 
 #define NVRAM_FILE	1
 
--- a/drivers/staging/brcm80211/util/siutils.c
+++ b/drivers/staging/brcm80211/util/siutils.c
@@ -69,7 +69,7 @@ static uint BCMINITFN(socram_banksize) (
 					uint8 idx, uint8 mtype);
 
 /* global variable to indicate reservation/release of gpio's */
-static uint32 si_gpioreservation = 0;
+static uint32 si_gpioreservation;
 
 /* global flag to prevent shared resources from being initialized multiple times in si_attach() */
 
